### PR TITLE
IDL changes for pinning

### PR DESCRIPTION
**Pull request by safmswork**
## Summary of the pull request
Adding stuff to the Devhome SDK IDL to support pinning extensions to the shell.

## Detailed description of the pull request / Additional comments
This is the first change needed for the upcoming Start Menu / Taskbar pinning of Dev Environments. This change adds a new interface with 6 new methods to enable pin/unpin operations , as well as a new runtimeclass and new enum values to support these methods. These changes have gone through the API change review and have been approved.

## Validation steps performed
Built the DevhomeSDK sln.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by EricJohnson327** [comment]
Please also update the sdkversion defined in build/azure-pipelines.yml to be 0.400

**Comment by safmswork** [comment]
> Please also update the sdkversion defined in build/azure-pipelines.yml to be 0.400
Done.  Also added comment in the PR description that this IDL change has gone through API review and been approved.

File: extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl [hunk 1]
```diff
@@ -578,6 +578,8 @@ namespace Microsoft.Windows.DevHome.SDK
         DeleteSnapshot = 0x00000400,
         ApplyConfiguration = 0x00000800,
         ModifyProperties = 0x00001000,
+        [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 4)] PinToStartMenu = 0x00002000,
+        [contract(Microsoft.Windows.DevHome.SDK.DevHomeContract, 4)] PinToTaskbar = 0x00004000,
```
Comment (bbonaby) [inline extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl:582]: I know this is probably just VS idl formatting but would it be possible to put these on a new line, the new enums I mean

**Review by huzaifa-d — APPROVED** [review]

**Review by bbonaby — APPROVED** [review]

File: extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl (hunk 1 above)
Comment (safmswork) [reply extensionsdk/Microsoft.Windows.DevHome.SDK/Microsoft.Windows.DevHome.SDK.idl]: Yep, I've changed it. That does look better

**Pull request merged**